### PR TITLE
Add a repro test for `IsCompatibleType(cseLclVarTyp, expTyp)`

### LINKED ISSUE #35724: Crossgen2: JIT assert "IsCompatibleType(cseLclVarTyp, expTyp)" in JIT\SIMD\SimpleSIMDProgram\SimpleSIMDProgram.cmd
Example run:

https://dev.azure.com/dnceng/public/_build/results?buildId=626338&view=ms.vss-test-web.build-test-results-tab&runId=19563378&resultId=102082&paneView=debug

OS: All
Architecture: x64 (the only one currently supported by Crossgen2 CI runs)

Call stack:

<pre>
>	clrjitilc.dll!assertAbort(const char * why, const char * file, unsigned int line) Line 312	C++
 	clrjitilc.dll!noWayAssertAbortHelper(const char * cond, const char * file, unsigned int line) Line 501	C++
 	clrjitilc.dll!noWayAssertBody(const char * cond, const char * file, unsigned int line) Line 527	C++
 	clrjitilc.dll!noWayAssertBodyConditional(const char * cond, const char * file, unsigned int line) Line 511	C++
 	clrjitilc.dll!CSE_Heuristic::PerformCSE(CSE_Heuristic::CSE_Candidate * successfulCandidate) Line 2775	C++
 	clrjitilc.dll!CSE_Heuristic::ConsiderCandidates() Line 3047	C++
 	clrjitilc.dll!Compiler::optValnumCSE_Heuristic() Line 3148	C++
 	clrjitilc.dll!Compiler::optOptimizeValnumCSEs() Line 3190	C++
 	clrjitilc.dll!Compiler::optOptimizeCSEs() Line 3556	C++
 	clrjitilc.dll!CompilerPhase::DoPhase() Line 96	C++
 	clrjitilc.dll!Phase::Run() Line 63	C++
 	[Inline Frame] clrjitilc.dll!DoPhase(Compiler *) Line 108	C++
 	clrjitilc.dll!Compiler::compCompile(void * * methodCodePtr, unsigned long * methodCodeSize, JitFlags * compileFlags) Line 4728	C++
 	clrjitilc.dll!Compiler::compCompileHelper(CORINFO_MODULE_STRUCT_ * classPtr, ICorJitInfo * compHnd, CORINFO_METHOD_INFO * methodInfo, void * * methodCodePtr, unsigned long * methodCodeSize, JitFlags * compileFlags) Line 6073	C++
 	[Inline Frame] clrjitilc.dll!Compiler::compCompile::__l46::__Body::Run(Compiler::compCompile::__l2::__JITParam *) Line 5408	C++
 	clrjitilc.dll!Compiler::compCompile(CORINFO_METHOD_STRUCT_ * methodHnd, CORINFO_MODULE_STRUCT_ * classPtr, ICorJitInfo * compHnd, CORINFO_METHOD_INFO * methodInfo, void * * methodCodePtr, unsigned long * methodCodeSize, JitFlags * compileFlags) Line 5412	C++
 	[Inline Frame] clrjitilc.dll!jitNativeCode::__l8::__Body::Run::__l6::__Body::Run(jitNativeCode::__l8::__Body::Run::__l5::__JITParam *) Line 6711	C++
 	clrjitilc.dll!`jitNativeCode'::`8'::__Body::Run(jitNativeCode::__l2::__JITParam * __JITpParam) Line 6715	C++
 	clrjitilc.dll!jitNativeCode(CORINFO_METHOD_STRUCT_ * methodHnd, CORINFO_MODULE_STRUCT_ * classPtr, ICorJitInfo * compHnd, CORINFO_METHOD_INFO * methodInfo, void * * methodCodePtr, unsigned long * methodCodeSize, JitFlags * compileFlags, void * inlineInfoPtr) Line 6739	C++
 	clrjitilc.dll!CILJit::compileMethod(ICorJitInfo * compHnd, CORINFO_METHOD_INFO * methodInfo, unsigned int flags, unsigned char * * entryAddress, unsigned long * nativeSizeOfCode) Line 307	C++
 	jitinterface.dll!JitCompileMethod(CorInfoException * * ppException, Jit * pJit, void * thisHandle, void * * callbacks, void * methodInfo, unsigned int flags, void * entryAddress, void * nativeSizeOfCode) Line 86	C++
 	[Managed to Native Transition]	
 	ILCompiler.ReadyToRun.dll!Internal.JitInterface.CorInfoImpl.CompileMethodInternal(ILCompiler.DependencyAnalysis.IMethodNode methodCodeNodeNeedingCode, Internal.IL.MethodIL methodIL) Line 164	C#
 	ILCompiler.ReadyToRun.dll!Internal.JitInterface.CorInfoImpl.CompileMethod(ILCompiler.DependencyAnalysis.ReadyToRun.MethodWithGCInfo methodCodeNodeNeedingCode) Line 218	C#
</pre>

Method being compiled: <code>SIMDDebugTest.Program+Vector2Test+DivTest.RunTests()</code>

NGenDump:

https://gist.github.com/trylek/b53f87d9e01f5c25766cc347f0032478

### PR DESCRIPTION
**Pull request by sandreenko**
Closes #35724.

The issue was fixed by #38484 (we don't longer try to CSE `IND SIMD8` there), that PR adds a repro test that does not require crossgen2.

### CODE REVIEW

**Comment by sandreenko** [comment]
This PR adds a repro test for a fixed issue.
PTAL @dotnet/jit-contrib

**Review by BruceForstall — APPROVED** [review]
Seems like you could make this a Pri-1 test.

**Comment by sandreenko** [comment]
> Seems like you could make this a Pri-1 test.

Thanks, done.

File: src/tests/JIT/Regression/JitBlue/Runtime_35144/Runtime_35144.csproj [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
+    <CLRTestPriority>1</CLRTestPriority>
```
Comment (drieseng) [inline src/tests/JIT/Regression/JitBlue/Runtime_35144/Runtime_35144.csproj:4]: Sorry if I totally miss the ball here, but wasn't **Runtime_35724** supposed to be marked as prio 1 ?

Comment (sandreenko) [reply src/tests/JIT/Regression/JitBlue/Runtime_35144/Runtime_35144.csproj:4]: yes, it was fixed in https://github.com/dotnet/runtime/pull/39062

**Pull request merged**